drivers/virtio/mmio/virtio_mmio_cmdl.c

```diff
@@ -0,0 +1,135 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <inttypes.h>
+
+#include <uk/alloc.h>
+#include <uk/bus/platform.h>
+#include <uk/libparam.h>
+#include <uk/essentials.h>
+
+#include <virtio/virtio_mmio.h>
+
+#define MAX_DEV_CMDLINE		CONFIG_VIRTIO_MMIO_MAX_DEV_CMDLINE
+
+static struct uk_alloc *virtio_mmio_cmdl_alloc;
+
+static char *uk_libparam_devices[MAX_DEV_CMDLINE];
+
+UK_LIBPARAM_PARAM_ARR_ALIAS(device, uk_libparam_devices, charp,
+			    MAX_DEV_CMDLINE, "virtio-mmio devices");
+
+/* Helper to parse numeric string with size suffix (eg "4K") into numeric */
+static char *parse_size(const char *size_str, size_t *sz)
+{
+	char *endptr;
+
+	*sz = strtoull(size_str, &endptr, 0);
+
+	switch (*endptr) {
+	case 'k':
+	case 'K':
+		*sz *= 1024;
+		++endptr;
+		break;
+	case 'm':
+	case 'M':
+		*sz *= 1024 * 1024;
+		++endptr;
+		break;
+	case 'g':
+	case 'G':
+		*sz *= 1024 * 1024 * 1024;
+		++endptr;
+		break;
+	default:
+		break;
+	}
```

Comment:
```suggestion
	switch (*endptr) {
	case 'g':
	case 'G':
		*sz *= 1024;
		/* fall through */
	case 'm':
	case 'M':
		*sz *= 1024;
		/* fall through */
	case 'k':
	case 'K':
		*sz *= 1024;
		++endptr;
		break;
	default:
		break;
	}
```
But I guess this is a matter of preference.

Reply:
Yes, I have seen this pattern before. We can do it that way too, either way looks fine to me

Reply:
And by the way, any objections on adding a `__fallthrough` definition in `essentials.h`?